test/test_torch.py

```diff
@@ -15465,14 +15466,34 @@ def run_subtest(matrix_size, batches, device, pivot, singular=False, a=None):
                 self.assertEqual(a_LU, a_LU_info)
                 self.assertEqual(pivots_info, pivots)
 
+                a = a.view(-1, a.size(-2), a.size(-1))
```

Comment:
I think we should keep `self.assertEqual(P.matmul(L.matmul(U)), a)` for non-complex dtype test and add:
```
if dtype.is_complex():
 TODO(@userid): remove this once bmm_out is implemented for CUDA complex types
 ...
else:
 ...
```